src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int NOT_EXISTS = -1;
+    private static final int UNIT = 1;
+    private T[] internalArray;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        capacity = 10;
+        internalArray = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < capacity) {
+            internalArray[size] = value;
+            size++;
+        } else {
+            growIfArrayFull();
+            add(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == index) {
+            add(value);
+        } else if (isIndexAcceptable(index)) {
+            size++;
+            if (size == capacity) {
+                growIfArrayFull();
+            }
+            System.arraycopy(internalArray, index, internalArray,
+                    index + UNIT, size - index - UNIT);
+            internalArray[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Unavailable index "
+                    + "for adding element!!!");
+        }
```

Comment:
I think, this part can be simplified a bit. And if you create a method for exception, the code becomes even clearer.
```suggestion
        if (size == index) {
            add(value);
            return;
        } 
        if (!isIndexAcceptable(index)) {
            throw new ArrayListIndexOutOfBoundsException("Unavailable index "
                    + "for adding element!!!");
        }
        size++;
        if (size == capacity) {
            growIfArrayFull();
        }
        System.arraycopy(internalArray, index, internalArray,
                index + UNIT, size - index - UNIT);
        internalArray[index] = value;
```